logging/CMakeLists.txt

```diff
@@ -1,114 +1,62 @@
-        set(LOG4CPLUS_INCLUDE_DIRS ${PC_LOG4CPLUS_INCLUDE_DIRS}  )
-      endif()
-    endif()
-
-    if (NOT PC_LOG4CPLUS_FOUND)
-      message(STATUS "Trying to find log4cplus without PkgConfig")
-      find_path(LOG4CPLUS_INCLUDE_DIRS log4cplus/config.hxx
-              HINTS ${CMAKE_INSTALL_PREFIX}
-              PATH_SUFFIXES include)
-
-      find_library(LOG4CPLUS_LIBRARIES NAMES log4cplus
-              HINTS ${CMAKE_INSTALL_PREFIX}
-              PATH_SUFFIXES lib)
-
-      include(FindPackageHandleStandardArgs)
-      find_package_handle_standard_args(LOG4CPLUS DEFAULT_MSG LOG4CPLUS_LIBRARIES LOG4CPLUS_INCLUDE_DIRS)
-    endif()
-  endif()
-  #######################################################################################################################
-  # Default CONCORD_LOGGER_NAME to "concord"
-
-  if(NOT CONCORD_LOGGER_NAME)
-    set(CONCORD_LOGGER_NAME "concord" CACHE STRING "Set concord logger name" FORCE)
-  endif()
-
-  #######################################################################################################################
-  #export all the way up to the top level directory
-
-  set(current_dir ${CMAKE_CURRENT_SOURCE_DIR})
-  while(NOT current_dir STREQUAL "")
-    message(STATUS "setting logger compile definitions for [${current_dir}]")
-    if(USE_LOG4CPP)
-      set_property(DIRECTORY ${current_dir} APPEND PROPERTY COMPILE_DEFINITIONS USE_LOG4CPP)
-      set_property(DIRECTORY ${current_dir} APPEND PROPERTY LINK_LIBRARIES      ${LOG4CPLUS_LIBRARIES})
-      set_property(DIRECTORY ${current_dir} APPEND PROPERTY INCLUDE_DIRECTORIES ${LOG4CPLUS_INCLUDE_DIRS})
-    endif()
-    set_property(DIRECTORY ${current_dir} APPEND PROPERTY COMPILE_DEFINITIONS DEFAULT_LOGGER_NAME="${CONCORD_LOGGER_NAME}")
-    set_property(DIRECTORY ${current_dir} APPEND PROPERTY INCLUDE_DIRECTORIES "$<TARGET_PROPERTY:logging,INTERFACE_INCLUDE_DIRECTORIES>")
-    set_property(DIRECTORY ${current_dir} APPEND PROPERTY COMPILE_OPTIONS "-Wformat=2")
-    set_property(DIRECTORY ${current_dir} APPEND PROPERTY COMPILE_OPTIONS "-Wno-deprecated") # TODO delete when upgrading to a newer version
-    get_property(current_dir DIRECTORY ${current_dir} PROPERTY PARENT_DIRECTORY)
-  endwhile()
-
-  #######################################################################################################################
-  #default logger initialization for tests
-
-  add_library(logging_dev OBJECT src/Logger.cpp)
-  #current version of log4cplus uses deprecated std::auto_ptr
-  target_compile_options(logging_dev PUBLIC -Wno-deprecated-declarations)
-  target_include_directories(logging_dev PUBLIC include/)
-endif()
-
+project(logging  LANGUAGES CXX)
+
+
+add_library(logging STATIC src/Logger.cpp )
+target_include_directories(logging PUBLIC include/)
+set_property(TARGET logging PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+if(USE_LOG4CPP)
+	message(STATUS "USE_LOG4CPP")
+	target_sources(logging PRIVATE src/Logging4cplus.cpp)
+
+	if (USE_CONAN)
```

Comment:
WE can probably remove all the conan stuff.

Reply:
Next step